test/functional/rpc_rawtransaction.py

```diff
@@ -69,24 +74,129 @@ def setup_network(self):
         super().setup_network()
         self.connect_nodes(0, 2)
 
+    def sync_peers(self):
+        self.sync_mempools(self.nodes[0:3])
+        self.sync_blocks()
+
+    def generate_and_sync(self, node, blocks):
+        self.sync_peers()
+        self.nodes[node].generate(blocks)
+        self.sync_peers()
+
     def run_test(self):
-        self.log.info('prepare some coins for multiple *rawtransaction commands')
-        self.nodes[2].generate(1)
-        self.sync_all()
-        self.nodes[0].generate(COINBASE_MATURITY + 1)
-        self.sync_all()
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(),1.5)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(),1.0)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(),5.0)
-        self.sync_all()
-        self.nodes[0].generate(5)
-        self.sync_all()
-
-        self.log.info('Test getrawtransaction on genesis block coinbase returns an error')
+        self.log.info("Prepare some coins for multiple *rawtransaction commands")
+        self.generate_and_sync(node=2, blocks=1)
+        self.generate_and_sync(node=0, blocks=COINBASE_MATURITY + 1)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
+        self.generate_and_sync(node=0, blocks=5)
+
+        self.getrawtransaction_tests()
+        self.createrawtransaction_tests()
+        self.sendrawtransaction_tests()
+        self.sendrawtransaction_testmempoolaccept_tests()
+        self.decoderawtransaction_tests()
+        self.transaction_version_number_tests()
+        if not self.options.descriptors:
+            self.raw_multisig_transaction_legacy_tests()
+
+    def getrawtransaction_tests(self):
+        addr = self.nodes[1].getnewaddress()
+        txid = self.nodes[0].sendtoaddress(addr, 10)
+        self.generate_and_sync(node=0, blocks=1)
+        vout = find_vout_for_address(self.nodes[1], txid, addr)
+        rawTx = self.nodes[1].createrawtransaction([{'txid': txid, 'vout': vout}], {self.nodes[1].getnewaddress(): 9.999})
+        rawTxSigned = self.nodes[1].signrawtransactionwithwallet(rawTx)
+        txId = self.nodes[1].sendrawtransaction(rawTxSigned['hex'])
```

Comment:
Not originally your code, but variables named `txid` and `txId` are very easy to confuse, I think.

Reply:
I initially planned to respond that this was out of scope, then saw this very issue had tripped me up (https://github.com/bitcoin/bitcoin/pull/22437#discussion_r684653116). Agree! Appended a final commit to change them to `txid1`, `txid2` when `txId` was being reused/reassigned in the same test function. Doing this also revealed an unused `txId` assignment; fixed.